content/posts/2021/10/libplanet-0.11-0.19/index.eng.md

```diff
@@ -0,0 +1,251 @@
+
+As of Libplanet [0.19.0], the current protocol version is 2, and any blocks
+produced before Libplanet [0.11.0], which introduced protocol versions,
+are treated as the protocol version 0.
+
+[^2]: This does not mean that all major and minor releases increase the protocol
+      version. In fact, most releases do not increment the protocol version.
+
+[Block<T>.ProtocolVersion]: https://docs.libplanet.io/0.19.0/api/Libplanet.Blocks.Block-1.html#Libplanet_Blocks_Block_1_ProtocolVersion
+
+
+Network Transport Layer
+-----------------------
+
+*Added in version [0.11.0].*
+
+Since its inception, Libplanet has left much of the low-level networking to
+[NetMQ], which is a .NET implementation of [ZeroMQ].  However, now it is
+necessary to change its networking layer to be possible without relying on NetMQ
+so that Libplanet can be used on various platforms such as mobile and Web.
+
+The network transport layer is Libplanet's response to that need.
+This abstraction layer was added experimentally to diversify low-level
+network transports and to allow game developers to define and use them
+if there are special requirements.
+
+API-wise, it is represented as the [`ITransport` interface][ITransport], and
+the [`NetMQTransport`][NetMQTransport] which implements it is built-in.
+
+As it is a pilot phase, the interface still has been adjusted through several
+minor releases, and game programmers can't yet use their own transport
+implementation.  Later in the year, [`Swarm<T>`][Swarm<T>] instances will be
+configurable with custom transports besides
+the [new TCP-based transport][TcpTransport], which aims to replace
+the `NetMQTransport`.
+
+[ZeroMQ]: https://zeromq.org/
+[NetMQ]: https://github.com/zeromq/netmq
+[ITransport]: https://docs.libplanet.io/0.19.0/api/Libplanet.Net.Transports.ITransport.html
+[NetMQTransport]: https://docs.libplanet.io/0.19.0/api/Libplanet.Net.Transports.NetMQTransport.html
+[TcpTransport]: https://github.com/planetarium/libplanet/pull/1523
+[Swarm<T>]: https://docs.libplanet.io/0.19.0/api/Libplanet.Net.Swarm-1.html
+
+
+Configurable <abbr title="Proof-Of-Work">PoW</abbr> Hash Algorithm
+------------------------------------------------------------------
+
+*Added in version [0.12.0].*
+
+Previously, Libplanet has used [SHA-256] for proof-of-work.  However, since
+version [0.12.0], each network can configure its hash algorithm for
+<abbr title="proof-of-work">PoW</abbr> by implementing its
+[`IBlockPolicy<T>.GetHashAlgorithm()`
+method][IBlockPolicy<T>.GetHashAlgorithm].  As it is not only configurable
+for each network, but also for block indices, it can be migrated within
+a network after its establishment.
+
+You can use any hash algorithm for <abbr title="proof-of-work">PoW</abbr>
+as long as it is a subclass of the [`HashAlgorithm`][HashAlgorithm] abstract
+class, which is built in the .NET runtime.  It is not necessary to be included
+in the .NET runtime, but you can use your custom implementation as well.
```

Comment:
```suggestion
class, which is built in the .NET runtime.  Since the hash algorithm is not necessarily required to be included
in the .NET runtime, you can use your custom implementation as well.
```